### PR TITLE
Add example for OCT()

### LINKED ISSUE #16940: Enrich instructions for string functions  NOT REGEXP and OCT()

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Closes #16940

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@dveeden Could you please invite a tech reviewer? Thanks!

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -1180,11 +1180,52 @@ Negation of simple pattern matching.
...
+Negation of [`REGEXP`](#regexp).
 
 ### [`OCT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_oct)
 
-Return a string containing octal representation of a number.
+Return a string containing octal (base 7) representation of a number.
+
+Examples:
+
+In the example below you can see that the decimal values 1 to 7 are identical to the octal numbers. The numbers 8 until 15 in decimal match with 10 to 17 octal.
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:1191]:
```suggestion
The following example generates a sequence of numbers from 1 to 20 using a [recursive common table expression (CTE)](/develop/dev-guide-use-common-table-expression.md#recursive-cte) and then uses the `OCT()` function to convert each number to its octal representation. Decimal values from 1 to 7 have identical representations in octal. Decimal numbers from 8 to 15 correspond to octal numbers from 10 to 17.
```

**Review by Oreoxmt — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -1180,11 +1180,52 @@ Negation of simple pattern matching.
...
+Negation of [`REGEXP`](#regexp).
 
 ### [`OCT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_oct)
 
-Return a string containing octal representation of a number.
+Return a string containing octal (base 7) representation of a number.
+
+Examples:
+
+The following example generates a sequence of numbers from 1 to 20 using a [recursive common table expression (CTE)](/develop/dev-guide-use-common-table-expression.md#recursive-cte) and then uses the `OCT()` function to convert each number to its octal representation. Decimal values from 1 to 7 have identical representations in octal. Decimal numbers from 8 to 15 correspond to octal numbers from 10 to 17.
```
Comment (mjonss) [inline functions-and-operators/string-functions.md:1191]:
```suggestion
The following example generates a sequence of numbers from 1 to 20 using a [recursive common table expression (CTE)](/develop/dev-guide-use-common-table-expression.md#recursive-cte) and then uses the `OCT()` function to convert each number to its octal representation. Decimal values from 0 to 7 have identical representations in octal. Decimal numbers from 8 to 15 correspond to octal numbers from 10 to 17, and so on.
```

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -1180,11 +1180,52 @@ Negation of simple pattern matching.
...
+
+```sql
+WITH RECURSIVE nr(n) AS (
+    SELECT 1 AS n
```
Comment (mjonss) [inline functions-and-operators/string-functions.md:1195]:
```suggestion
    SELECT 0 AS n
```

File: functions-and-operators/string-functions.md [hunk 4]
```diff
@@ -1180,11 +1180,52 @@ Negation of simple pattern matching.
...
+    SELECT n+1 FROM nr WHERE n<20
+)
+SELECT n, OCT(n) FROM nr;
+```
+
+```
++------+--------+
+| n    | OCT(n) |
++------+--------+
+|    1 | 1      |
```
Comment (mjonss) [inline functions-and-operators/string-functions.md:1206]:
```suggestion
|    0 | 0      |
|    1 | 1      |
```

**Review by mjonss — APPROVED** [review]
Looks good with one fix (octal is base 8).

File: functions-and-operators/string-functions.md [hunk 5]
```diff
@@ -1180,11 +1180,53 @@ Negation of simple pattern matching.
...
+Negation of [`REGEXP`](#regexp).
 
 ### [`OCT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_oct)
 
-Return a string containing octal representation of a number.
+Return a string containing [octal](https://en.wikipedia.org/wiki/Octal) (base 8) representation of a number.
+
+Examples:
+
+The following example generates a sequence of numbers from 1 to 20 using a [recursive common table expression (CTE)](/develop/dev-guide-use-common-table-expression.md#recursive-cte) and then uses the `OCT()` function to convert each number to its octal representation. Decimal values from 0 to 7 have identical representations in octal. Decimal numbers from 8 to 15 correspond to octal numbers from 10 to 17, and so on.
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:1191]:
```suggestion
The following example generates a sequence of numbers from 1 to 20 using a [recursive common table expression (CTE)](/develop/dev-guide-use-common-table-expression.md#recursive-cte) and then uses the `OCT()` function to convert each number to its octal representation. Decimal values from 0 to 7 have identical representations in octal. Decimal numbers from 8 to 15 correspond to octal numbers from 10 to 17.
```

https://developers.google.com/style/word-list#and-so-on
> Avoid using etc., and so forth, and and so on wherever possible.

**Review by Oreoxmt — APPROVED** [review]

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**